tests/core/object/test_undo_redo.h

```diff
@@ -0,0 +1,211 @@
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_UNDO_REDO_H
+#define TEST_UNDO_REDO_H
+
+#include "core/object/undo_redo.h"
+#include "tests/test_macros.h"
+
+// Declared in global namespace because of GDCLASS macro warning (Windows):
+// "Unqualified friend declaration referring to type outside of the nearest enclosing namespace
+// is a Microsoft extension; add a nested name specifier".
+class _TestUndoRedoObject : public Object {
+	GDCLASS(_TestUndoRedoObject, Object);
+	int property_value = 0;
+
+protected:
+	static void _bind_methods() {
+		ClassDB::bind_method(D_METHOD("set_property", "property"), &_TestUndoRedoObject::set_property);
+		ClassDB::bind_method(D_METHOD("get_property"), &_TestUndoRedoObject::get_property);
+		ADD_PROPERTY(PropertyInfo(Variant::INT, "property"), "set_property", "get_property");
+
+		ClassDB::bind_method(D_METHOD("add_to_property"), &_TestUndoRedoObject::add_to_property);
+		ClassDB::bind_method(D_METHOD("subtract_from_property"), &_TestUndoRedoObject::subtract_from_property);
+	}
+
+public:
+	void set_property(int value) { property_value = value; }
+	int get_property() const { return property_value; }
+	void add_to_property(int value) { property_value += value; }
+	void subtract_from_property(int value) { property_value -= value; }
+};
+
+namespace TestUndoRedo {
+
+void set_property_action(UndoRedo *undo_redo, const String &name, _TestUndoRedoObject *test_object, int value, UndoRedo::MergeMode merge_mode = UndoRedo::MERGE_DISABLE) {
+	undo_redo->create_action(name, merge_mode);
+	undo_redo->add_do_property(test_object, "property", value);
+	undo_redo->add_undo_property(test_object, "property", test_object->get_property());
+	undo_redo->commit_action();
+}
+
+void increment_property_action(UndoRedo *undo_redo, const String &name, _TestUndoRedoObject *test_object, int value, UndoRedo::MergeMode merge_mode = UndoRedo::MERGE_DISABLE) {
+	undo_redo->create_action(name, merge_mode);
+
+	Variant val = Variant(value);
+	const Variant *argptrs[1];
+	argptrs[0] = &val;
+
+	undo_redo->add_do_method(Callable(test_object, "add_to_property").bindp(argptrs, 1));
+	undo_redo->add_undo_method(Callable(test_object, "subtract_from_property").bindp(argptrs, 1));
+
+	undo_redo->commit_action();
+}
+
+TEST_CASE("[UndoRedo] Simple Property UndoRedo") {
+	GDREGISTER_CLASS(_TestUndoRedoObject);
+	UndoRedo *undo_redo = memnew(UndoRedo());
+
+	_TestUndoRedoObject *test_object = memnew(_TestUndoRedoObject());
+
+	CHECK(test_object->get_property() == 0);
```

Comment:
This check looks redundant. It's a test object which was just created, so obviously the value will be correct.